src/framebuffer.ml

```diff
@@ -197,6 +201,7 @@ let rightmost span =
   | Pair (_, x1) -> x1
 
 let filled_triangle (x0 : int) (y0 : int) (x1 : int) (y1 : int) (x2 : int) (y2 : int) (col : int) (buffer : t) =
+  buffer.dirty <- true;
```

Comment:
Again, this feels too early: What if the triangle was too small to draw? Currently looking at the code I suspect we always draw one pixel if the triangle is implossibly small, but if we ever fixed that, then I imagine we'd get into trouble. I'd feel much better if this change was next to the lines that actually update the buffer.